qemu/tests/virtio_console.py

```diff
@@ -1805,6 +1807,14 @@ def test_failed_boot():
         :param cfg: virtio_console_params - Expected error message.
         """
         exp_error_message = params['virtio_console_params']
+        version_params = params["version_params"]
+        qemu_binary = utils_misc.get_qemu_binary(params)
+        output = str(process.run(qemu_binary + " --version"))
+        re_comp = re.compile(r'\d+.\d+.\d+-\d+..+')
+        output_list = re_comp.findall(output)
+        if not re.search(version_params, output_list[0]):
+            params["extra_params"] = params.get("extra_params_rhev")
+            exp_error_message = params.get("virtio_console_params_rhev")
```

Comment:
It does not matter using rhev or rhel. I mean if you write as follows(you need to modify value in cfg accordingly), the code efficiency will be a little higher. Because we mainly test rhev, this can avoid getting "extra_params" and "exp_error_message" twice.
+        version_params = params["version_params"]
 +        qemu_binary = utils_misc.get_qemu_binary(params)
 +        output = str(process.run(qemu_binary + " --version"))
 +        re_comp = re.compile(r'\d+.\d+.\d+-\d+..+')
 +        output_list = re_comp.findall(output)
##The above is for rhev
 +        if  **re.search**(version_params, output_list[0]):
##Only for rhel, the following code will be executed
 +            params["extra_params"] = params.get("extra_params_**rhel"**)
 +            exp_error_message = params.get("virtio_console_params_**rhel**")